sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/cardinality/ApproximateDistinct.java

```diff
@@ -0,0 +1,352 @@
+      HyperLogLogPlus mergedAccum = createAccumulator();
+      for (HyperLogLogPlus accum : accumulators) {
+        try {
+          mergedAccum.addAll(accum);
+        } catch (CardinalityMergeException e) {
+          // Should never happen because only HyperLogLogPlus accumulators are instantiated.
+          LOG.error("The accumulators cannot be merged : " + e.getMessage(), e);
+        }
+      }
+      return mergedAccum;
+    }
+
+    @Override
+    public Coder<HyperLogLogPlus> getAccumulatorCoder(CoderRegistry registry, Coder inputCoder) {
+      return HyperLogLogPlusCoder.of();
+    }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+        super.populateDisplayData(builder);
+        builder
+                .add(DisplayData.item("p", p)
+                        .withLabel("Precision"))
+                .add(DisplayData.item("sp", sp)
+                        .withLabel("Sparse Representation precision"));
+      }
+  }
+
+  static class HyperLogLogPlusCoder extends CustomCoder<HyperLogLogPlus> {
+
+    private static final HyperLogLogPlusCoder INSTANCE = new HyperLogLogPlusCoder();
+
+    private static final ByteArrayCoder BYTE_ARRAY_CODER = ByteArrayCoder.of();
+
+    public static HyperLogLogPlusCoder of() {
+      return INSTANCE;
+    }
+
+    @Override public void encode(HyperLogLogPlus value, OutputStream outStream) throws IOException {
+      if (value == null) {
+        throw new CoderException("cannot encode a null HyperLogLogPlus sketch");
+      }
+      BYTE_ARRAY_CODER.encode(value.getBytes(), outStream);
+    }
+
+    @Override public HyperLogLogPlus decode(InputStream inStream) throws IOException {
+      return HyperLogLogPlus.Builder.build(BYTE_ARRAY_CODER.decode(inStream));
+    }
+
+    @Override public boolean isRegisterByteSizeObserverCheap(HyperLogLogPlus value) {
+      return true;
+    }
+
+    @Override protected long getEncodedElementByteSize(HyperLogLogPlus value) throws IOException {
+      if (value == null) {
+          throw new CoderException("cannot encode a null HyperLogLogPlus sketch");
+      }
+      return value.sizeof();
+    }
+  }
+
+    /**
+     * Computes the precision based on the desired relative error.
+     *
+     * <p>According to the paper, the mean squared error is bounded by the following formula :
+     * <pre>b(m) / sqrt(m)
+     * Where m is the number of buckets used (p = log2(m) )
+     * and b(m) < 1.106 for m > 16 ( p > 4).
+     * </pre>
+     *
+     * <br><b>WARNING : </b>
+     * <br>This does not means relative error in the estimation <b>can't</b> be higher.
+     * <br>This only means that, on average, the relative error will be
+     * lower than the desired relative error.
+     * <br>Nevertheless, the more elements arrive in the stream, the lower the variation will be.
+     * <br>Indeed, this is like when you throw a dice thousands or millions of time :
```

Comment:
<!--thread_id:cc_131403640_t; commit:6bf51fdf6afe6f389415792f335dd08adf9fa85a; resolved:0-->
<!--section:context-quote-->
> **ArnaudFnr** wrote:
> I think you meant dice* but ok for millions, makes more sense.

<!--section:body-->
I believe using ```die``` is correct since you are throwing a singular die millions of times.

Also, drop the preceding spaces, ```time : -> time:```